src/main/java/sorald/annotations/ProcessorsClassGenerator.java

```diff
@@ -22,7 +22,7 @@
     public void process(ProcessorAnnotation annotation, CtClass<T> element) {
         if (cu == null) {
             cu = getFactory().createCompilationUnit();
-            processorsClass = getFactory().Type().get("sorald.Processors");
+            processorsClass = getFactory().Type().get("sorald.sonar.SonarProcessorRepository");
```

Comment:
We should load `processorsClass` using `ServiceLoader`.

Reply:
Oh, you want to load each processor with SPI?

Reply:
The code generator class would be required for all static analyzers, right? So I thought we could generalise this as well. What do you think?

Reply:
No, it's not needed for all analyzers. How other analyzers implement the ProcessorRepository API is an internal implementation detail. We could change the internals of our `ProcessorRepository` to use SPI instead of codegen.

Reply:
> How other analyzers implement the ProcessorRepository API is an internal implementation detail

That is a strong point. We use code generation it _implement_ SonarProcessor repository and we might need to for other static analyzers. 

Should we also put this class inside `sonar` package? Eventually, it will become a maven dependency so it will be clear which files we need to package.

Reply:
Yes, this class needs to be moved or replaced by SPI. It's more or less your choice.